third_party/move/move-compiler-v2/tests/checking/typing/recursive_constant.exp

```diff
@@ -1,9 +1,7 @@
-// -- Model dump before bytecode pipeline
-module 0x42::M {
-    public fun get_x(): u64 {
-        false
-    }
-    public fun get_y(): u64 {
-        false
-    }
-} // end 0x42::M
+
+Diagnostics:
+error: Found recursive definition of a constant
+  ┌─ tests/checking/typing/recursive_constant.move:4:5
+  │
+4 │     const Y: u64 = X;
```

Comment:
Could we add some more tests for the recursive check?
- self recursion
- recursion with a more complicated cycle involving expressions

Reply:
done